CMakeLists.txt

```diff
@@ -2,7 +2,7 @@
 # Defines fc library target.
 
 PROJECT( fc )
-CMAKE_MINIMUM_REQUIRED( VERSION 2.8.12 )
+CMAKE_MINIMUM_REQUIRED( VERSION 3.2 )
```

Comment:
Any specific reason for this bump?

Reply:
I do not encourage changes for no reason, but I do know that cmake is evolving. Based on the deprecation warnings I receive on some cmake versions, I predict we will soon not be able to build using newer versions. 

I strongly suggest we audit the cmake scripts to reduce/remove the deprecations. If that requires a bump in version to something reasonable, I am all for it.

I realize such an audit may be a bit out of the scope of this ticket, but I just wanted to express my view on cmake versions.

Reply:
Yes, because of the generators that CMake supports.
The oldest version that supports Visual Studio 2015 is CMake 3.1 and I went just a little bit further, to be sure.
 
 IMHO we should move away from the fairly old generation of CMake 2 for sure.

Reply:
@jmjatlanta, yes, this would probably "fix" all the policy warnings.